### PR TITLE
Ignore special files on deploy

### LINKED ISSUE #112: Named pipe prevents deployment
A fifo file in the root prevents deployment, likely waiting for fifo output. Suggest checking for filetype before reading contents.

### PR DESCRIPTION
**Pull request by OlliV**
Fixes #112

### CODE REVIEW

File: lib/get-files.js [hunk 1]
```diff
@@ -135,13 +143,20 @@ export async function docker(path, {
...
     const relativePath = file.substr(prefixLength)
 
     if (relativePath === '') {
       return true
     }
 
+    const st = await stat(file)
+    const mode = new Mode(st)
+    if (!(mode.isDirectory() || mode.isFile() || mode.isSymbolicLink())) {
+      console.log('> [debug] ignoring special file "%s"', file)
```
Comment (matheuss) [inline lib/get-files.js]: @OlliV shouldn't this (and the other one too) be wrapped with `if (debug)`? 😅

Comment (OlliV) [reply lib/get-files.js]: thanks :D

File: lib/get-files.js [hunk 2]
```diff
@@ -135,13 +145,22 @@ export async function docker(path, {
   ).createFilter()
 
   const prefixLength = path.length + 1
-  const accepts = function (file) {
+  const accepts = async function (file) {
```
Comment (TooTallNate) [inline lib/get-files.js:148]: Seems like this `accepts` function should be DRY'd up a bit, but that's unrelated to this PR.

**Comment by TooTallNate** [comment]
LGTM for the bug fix.

**Review by rase- — APPROVED** [review]

**Comment by sugademo-1** [comment]
iam having this issue....plz help me to deploy
C:\Users\Ragu\Desktop\p>now
> Deploying ~\Desktop\p under tsuganya09@gmail.com
> Using Node.js 8.6.0 (default)
> Ready! https://progressive-jzdifagksr.now.sh [6s]
> You (tsuganya09@gmail.com) are on the OSS plan. Your code and logs will be made public.
> NOTE: You can use `now --public` to skip this prompt
> Initializing…
> Building
> ▲ npm install
> ✓ Using "package-lock.json"
> ⧗ Installing 13 main dependencies…
> ✓ Installed 636 modules [13s]
> ▲ npm start
> > progressive@1.0.0 start /home/nowuser/src
> > node ./index.js
> npm WARN lifecycle The node binary used for scripts is /home/nowuser/bin/node but npm is using /usr/bin/node itself. Use the `--scripts-prepend-node-path` option to include the path for the node binary npm was executed with.
> module.js:473
>       throw err;
>       ^
> Error: Cannot find module '/home/nowuser/src/index.js'
>     at Function.Module._resolveFilename (module.js:527:15)
>     at Function.Module._load (module.js:453:25)
>     at Function.Module.runMain (module.js:665:10)
>     at startup (bootstrap_node.js:187:16)
>     at bootstrap_node.js:607:3
> npm ERR! code ELIFECYCLE
> npm ERR! errno 1
> npm ERR! progressive@1.0.0 start: `node ./index.js`
> npm ERR! Exit status 1
> npm ERR!
> npm ERR! Failed at the progressive@1.0.0 start script.
> npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
> npm ERR! A complete log of this run can be found in:
> npm ERR!     /home/nowuser/.npm/_logs/2017-10-13T04_58_59_886Z-debug.log

**Pull request merged**